tests/mocks/Gateway.go

```diff
@@ -0,0 +1,263 @@
+// Code generated by mockery 2.7.5. DO NOT EDIT.
```

Comment:
does this generated code need to be committed?

Reply:
It doesn't and I plan to remove it, but I will do that after I setup GitHub actions to auto-generate it before running tests since it is required during the testing, but good flag